middleware/authorization.go

```diff
@@ -50,10 +54,28 @@ func PermissionCheck(next echo.HandlerFunc) echo.HandlerFunc {
 				return fmt.Errorf("error casting identity to struct: %+v", c.Get("identity"))
 			}
 
-			// current sources-api behavior = if there is a system key -> it's authorized.
-			// TODO: make this more specific and do more checks.
+			// checking to see if we're going to change the results since
```

Comment:
Maybe follow-up but do you think that this logic could be refactored somehow ?

it is well described but there is lot of ifs, switches - it can be hard understand later.

Reply:
Yeah I'm still playing with it - in the process of refactoring a bit still, I can do it in a followup pr though thats nbd.

Reply:
Ended up opening another PR after this one #66, this will probably still get refactored later.